service/src/main/java/org/sunbird/actor/organisation/validator/OrgTypeValidator.java

```diff
@@ -0,0 +1,162 @@
+package org.sunbird.actor.organisation.validator;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.collections.MapUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.sunbird.exception.ProjectCommonException;
+import org.sunbird.exception.ResponseCode;
+import org.sunbird.keys.JsonKey;
+import org.sunbird.logging.LoggerUtil;
+import org.sunbird.model.organisation.OrganisationType;
+
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Collectors;
+
+public class OrgTypeValidator {
+    private static OrgTypeValidator instance = null;
+    private ConcurrentMap<String, OrganisationType> orgTypeMap = null;
+    private LoggerUtil logger = new LoggerUtil(this.getClass());
+
+    private OrgTypeValidator() {
+    }
+
+    public static OrgTypeValidator getInstance() {
+        if (instance == null) {
+            instance = new OrgTypeValidator();
+        }
+        return instance;
+    }
+
+    public boolean isOrgTypeInitialized() {
+        return !MapUtils.isEmpty(orgTypeMap);
+    }
+
+    public void initializeOrgType(List<Map<String, String>> orgTypeConfig) {
+        ObjectMapper mapper = new ObjectMapper();
+        if (orgTypeMap == null) {
+            orgTypeMap = new ConcurrentHashMap<String, OrganisationType>();
+        } else {
+            orgTypeMap.clear();
+        }
+        for (Map<String, String> orgTypeString : orgTypeConfig) {
+            OrganisationType orgType = mapper.convertValue(orgTypeString, OrganisationType.class);
+            orgTypeMap.put(orgType.getName(), orgType);
+        }
+        try {
+            logger.info("OrgType Map is initialized. " + mapper.writeValueAsString(orgTypeMap));
```

Comment:
Add requestContext object for logging

Reply:
Updated code and added request context in logger